plugin.rb

```diff
@@ -16,5 +16,58 @@ module ::MyPluginModule
 require_relative "lib/my_plugin_module/engine"
 
 after_initialize do
-  # Code which should run after Rails has finished booting
+  reloadable_patch do
+    module ::TopicControllerNoIndexExtension
+      Discourse::Application.routes.prepend do
+        put "t/:topic_id/toggle-noindex" => "topics#toggle_noindex",
+            :constraints => {
+              topic_id: /\d+/,
+            }
+      end
+
+      def toggle_noindex
+        head :forbidden and return unless guardian.is_staff?
+        topic_id = params.require(:topic_id).to_i
+        begin
+          topic = Topic.find(topic_id)
+          current = topic.custom_fields["noindex"]
+          newval = nil
+          if current.nil? || current == "f"
+            newval = "t"
+          else
+            newval = "f"
+          end
+          topic.custom_fields["noindex"] = newval
+          topic.save!
+
+          render json: success_json
+        rescue ActiveRecord::RecordInvalid
+          render json: failed_json, status: 422
+        end
+      end
+
+      def show
+        super
+        response.headers["X-Robots-Tag"] = "noindex" if @topic_view.topic.noindex
+      end
+    end
+
+    ::TopicsController.prepend ::TopicControllerNoIndexExtension
+
+    require_dependency "topic"
+    class ::Topic
+      def noindex
+        custom_fields["noindex"] == "t"
+      end
+    end
+
+    add_to_serializer(:topic_view, :noindex) { object.topic.noindex }
+
+    require_dependency "topic_guardian"
+    module ::TopicGuardian
+      def can_change_topic_noindex?
+        is_staff?
+      end
+    end
```

Comment:
Is this actually being used anywhere? It looks like the controller is just doing a `staff?` check directly. (which IMO is fine - there's no hard requirement for the `guardian` indirection for this simple use case)